### PR TITLE
Select carousel chart

### PR DESCRIPTION
**Pull request by Hoan-K-Le**
Finish the basic functionality/css for line graph for now

### CODE REVIEW

File: app/components/Charts/LineChart.tsx [hunk 1]
```diff
@@ -28,34 +28,123 @@ const options = {
...
+    dates: number[];
+    current_date: string;
+  };
+  coinThree: {
+    current_price: number;
+    prices: string;
+    dates: number[];
+    current_date: string;
+  };
+};
```
Comment (yoloOnTheBattlefield) [inline app/components/Charts/LineChart.tsx:65]: you are repeating yourself here mate

File: app/components/Charts/LineChart.tsx [hunk 2]
```diff
@@ -28,34 +28,123 @@ const options = {
...
+    current_date: "",
+    dates: [],
+  },
+  coinThree: {
+    current_price: 0,
+    prices: "",
+    current_date: "",
+    dates: [],
+  },
+};
```
Comment (yoloOnTheBattlefield) [inline app/components/Charts/LineChart.tsx]: same here

File: app/components/CoinCarousel/index.tsx [hunk 3]
```diff
@@ -23,19 +32,27 @@ const CoinCarousel = () => {
...
+    } else if (newSelectCoin.length < 3) {
+      newSelectCoin.push(coin);
     }
+    setSelectCoin(newSelectCoin);
+    dispatch(updateSelectedCoinOne(newSelectCoin[0]?.id || ""));
+    dispatch(updateSelectCoinOneSymbol(newSelectCoin[0]?.symbol || ""));
+    dispatch(updateSelectedCoinTwo(newSelectCoin[1]?.id || ""));
+    dispatch(updateSelectCoinTwoSymbol(newSelectCoin[1]?.symbol || ""));
+    dispatch(updateSelectedCoinThree(newSelectCoin[2]?.id || ""));
+    dispatch(updateSelectCoinThreeSymbol(newSelectCoin[2]?.symbol || ""));
```
Comment (yoloOnTheBattlefield) [inline app/components/CoinCarousel/index.tsx]: you repeat too much here

File: app/components/Charts/LineChart.tsx [hunk 4]
```diff
@@ -64,70 +158,202 @@ export default function LineChart() {
...
+          prevState.map(c => {
+            if (c.name === coinName) {
+              return {
+                ...c,
+                current_price: getCurrentData.length
+                  ? getCurrentData[0][1]
+                  : null,
+                current_date: getCurrentData.length
+                  ? getCurrentData[0][0]
+                  : null,
```
Comment (yoloOnTheBattlefield) [inline app/components/Charts/LineChart.tsx]: can be done better

File: app/components/Charts/LineChart.tsx [hunk 5]
```diff
@@ -64,70 +158,202 @@ export default function LineChart() {
...
+                coin_name: "",
+              }
+            : c
+        )
+      );
+
+      for (const { coin, name } of coinsForFetching) {
+        await fetchChartData(coin, name);
+      }
+    };
```
Comment (yoloOnTheBattlefield) [inline app/components/Charts/LineChart.tsx:221]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

File: app/components/Charts/LineChart.tsx [hunk 6]
```diff
@@ -64,70 +158,202 @@ export default function LineChart() {
...
+      setGradients(prevState => ({
+        coinOne: {
+          gradient: gradient,
+        },
+        coinTwo: {
+          gradient: gradientTwo,
+        },
+        coinThree: {
+          gradient: gradientThree,
+        },
```
Comment (yoloOnTheBattlefield) [inline app/components/Charts/LineChart.tsx]: why are you using objects?

File: app/components/Charts/LineChart.tsx [hunk 7]
```diff
@@ -64,70 +158,202 @@ export default function LineChart() {
...
+      }));
     }
-  }, [currentCurrency]);
+  }, [currentCurrency, currentCoinOne, currentCoinTwo, currentCoinThree]);
+  useEffect(() => {
+    console.log(coins, "hello coins arr");
```
Comment (yoloOnTheBattlefield) [inline app/components/Charts/LineChart.tsx]: what's this?

File: app/components/Charts/LineChart.tsx [hunk 8]
```diff
@@ -64,70 +158,202 @@ export default function LineChart() {
...
+              <div className="flex items-center gap-2" key={coin.name}>
+                <div
+                  className={`h-[14px] w-[14px] rounded gap-2 ${
+                    coin.name === "coinOne"
+                      ? "bg-[#7878FA]"
+                      : coin.name === "coinTwo"
+                      ? "bg-[#D878FA]"
+                      : coin.name === "coinThree"
+                      ? "bg-green-300"
+                      : ""
```
Comment (yoloOnTheBattlefield) [inline app/components/Charts/LineChart.tsx]: is this Chalcatongo Mixtec?

File: app/store/store.ts [hunk 9]
```diff
@@ -4,13 +4,21 @@ import currencyReducer from "./CurrencyReducer";
 import coinsReducer from "./CoinsData";
 import globalCoinReducer from "./CoinsBarData";
 import coinGraphReducer from "./ChartSelectorData";
+import {
+  coinOneReducer,
+  coinTwoReducer,
+  coinThreeReducer,
```
Comment (yoloOnTheBattlefield) [inline app/store/store.ts]: you made a reducer for each coin?

**Review by yoloOnTheBattlefield — APPROVED** [review]

File: app/components/Charts/LineChart.tsx (hunk 1 above)
Comment (Hoan-K-Le) [reply app/components/Charts/LineChart.tsx:65]: I don't get this part, its a typescript type for the state?

File: app/components/Charts/LineChart.tsx (hunk 5 above)
Comment (Hoan-K-Le) [reply app/components/Charts/LineChart.tsx:221]: Is that what you're looking for? Or should I do it in a way there I create 3 seperate api call and return it in Promise.all([fetch1,fetch2,fetch3]) ?

**Review by yoloOnTheBattlefield — APPROVED** [review]

**Pull request merged**